src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -140,7 +140,20 @@ public virtual void Init(IProviderConfiguration providerCfg, string providerName
             {
                 var bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(adapterSettings.CacheSizeMb, () => new FixedSizeBuffer(1 << 20));
                 var timePurge = new TimePurgePredicate(adapterSettings.DataMinTimeInCache, adapterSettings.DataMaxAgeInCache);
-                CacheFactory = (partition,checkpointer,cacheLogger) => new EventHubQueueCache(checkpointer, bufferPool, timePurge, cacheLogger, this.SerializationManager);
+                if (adapterSettings.SlowConsumingMonitorThreshold > 0)
+                {
+                    CacheFactory = (partition, checkpointer, cacheLogger) =>
+                    {   var cache = new EventHubQueueCache(checkpointer, bufferPool, timePurge, cacheLogger, this.SerializationManager);
```

Comment:
suggest minimizing code in conditional statement.  Only thing that depend on the condition is if the SlowConsumingPressureMonitor is added. so

    var cache = new .....
    if(..)
    {
        add monitor
    }
    return cache;